Installation on osx-arm64

I am trying to install NaMaster from source on an Apple Silicon machine. I found from conda-forge that osx-arm64 support is not available. Running `pip install .`, I get error on dependencies. I had linked the `chealpix` and the healpix included `libsharp` to the `_deps\lib` and `_deps\include` subfolders, but dependencies are still failing. GSL, FFTW and cfitsio are all installed via `brew` and included in `PATH`. 

Full error:
`Building wheels for collected packages: pymaster
  Building wheel for pymaster (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [93 lines of output]
      running bdist_wheel
      running build
      running build_py
      Found automake.
      configure.ac:23: warning: AC_PROG_CC_C99 is obsolete; use AC_PROG_CC
      configure.ac:26: warning: ac_ext=c
      configure.ac:26: ac_cpp='$CPP $CPPFLAGS'
      configure.ac:26: ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
      configure.ac:26: ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
      configure.ac:26: ac_compiler_gnu=$ac_cv_c_compiler_gnu
      configure.ac:26: if test -n "$ac_tool_prefix"; then
      configure.ac:26:   # Extract the first word of "${ac_tool_prefix}gcc", so it can be a program name with args.
      configure.ac:26: set dummy ${ac_tool_prefix}gcc; ac_word=$2
      configure.ac:26: { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
      configure.ac:26: printf %s "checking for $ac_word... " >&6; }
      configure.ac:26: if test ${ac_cv_prog_CC+y}
      configure.ac:26: then :
      configure.ac:26:   printf %s "(cached) " >&6
      configure.ac:26: else $as_nop
      configure.ac:26:   if test -n "$CC"; then
      configure.ac:26:   ac_cv_prog_CC="$CC" # Let the user override the test.
      configure.ac:26: else
      configure.ac:26: as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
      configure.ac:26: for as_dir in $PATH
      configure.ac:26: do
      configure.ac:26:   IFS=$as_save_IFS
      configure.ac:26:   case $as_dir in #(((
      configure.ac:26:     '' is m4_require'd but not m4_defun'd
      lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
      lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
      ./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
      ./lib/autoconf/programs.m4:41: _AC_CHECK_PROG is expanded from...
      ./lib/autoconf/programs.m4:101: AC_CHECK_PROG is expanded from...
      ./lib/autoconf/programs.m4:221: AC_CHECK_TOOL is expanded from...
      ./lib/autoconf/c.m4:452: AC_PROG_CC is expanded from...
      configure.ac:26: the top level
      configure.ac:32: error: possibly undefined macro: AC_PROG_LIBTOOL
            If this token and others are legitimate, please use m4_pattern_allow.
            See the Autoconf documentation.
      autoreconf: error: /opt/homebrew/Cellar/autoconf/2.71/bin/autoconf failed with exit status: 1
      Using -march=native. Binary will not be portable.
      configure: WARNING: unrecognized options: --enable-shared, --with-pic
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a race-free mkdir -p... ./../../install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking whether to enable maintainer-specific portions of Makefiles... yes
      checking whether make supports the include directive... yes (GNU style)
      checking for arm64-apple-darwin20.0.0-gcc... arm64-apple-darwin20.0.0-clang
      checking whether the C compiler works... no
      configure: error: in `/Users/shamik/opt/NaMaster/_deps/libsharp2':
      configure: error: C compiler cannot create executables
      See `config.log' for more details
      make: *** No targets specified and no makefile found.  Stop.
      make: *** No rule to make target `install'.  Stop.
      cp: directory ../include/libsharp2 does not exist
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a thread-safe mkdir -p... ./install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking for style of include used by make... GNU
      checking for arm64-apple-darwin20.0.0-gcc... arm64-apple-darwin20.0.0-clang
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether arm64-apple-darwin20.0.0-clang accepts -g... yes
      checking for arm64-apple-darwin20.0.0-clang option to accept ISO C89... none needed
      checking whether arm64-apple-darwin20.0.0-clang understands -c and -o together... yes
      checking dependency style of arm64-apple-darwin20.0.0-clang... gcc3
      checking for arm64-apple-darwin20.0.0-ar... arm64-apple-darwin20.0.0-ar
      checking the archiver (arm64-apple-darwin20.0.0-ar) interface... ar
      checking build system type... Invalid configuration `arm64-apple-darwin20.0.0': machine `arm64-apple' not recognized
      configure: error: /bin/sh ./config.sub arm64-apple-darwin20.0.0 failed
      ERROR: failed to configure libnmt. Check all dependencies are installed
             Dependencies:
             - GSL
             - FFTW
             - CFITSIO
             - HEALPix
      error: Failed to compile C library.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pymaster
  Running setup.py clean for pymaster
Failed to build pymaster
Installing collected packages: pymaster
  Running setup.py install for pymaster ... error
  error: subprocess-exited-with-error

  × Running setup.py install for pymaster did not run successfully.
  │ exit code: 1
  ╰─> [97 lines of output]
      running install
      /Users/shamik/miniforge3/envs/arm/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      Found automake.
      configure.ac:23: warning: AC_PROG_CC_C99 is obsolete; use AC_PROG_CC
      configure.ac:26: warning: ac_ext=c
      configure.ac:26: ac_cpp='$CPP $CPPFLAGS'
      configure.ac:26: ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
      configure.ac:26: ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
      configure.ac:26: ac_compiler_gnu=$ac_cv_c_compiler_gnu
      configure.ac:26: if test -n "$ac_tool_prefix"; then
      configure.ac:26:   # Extract the first word of "${ac_tool_prefix}gcc", so it can be a program name with args.
      configure.ac:26: set dummy ${ac_tool_prefix}gcc; ac_word=$2
      configure.ac:26: { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
      configure.ac:26: printf %s "checking for $ac_word... " >&6; }
      configure.ac:26: if test ${ac_cv_prog_CC+y}
      configure.ac:26: then :
      configure.ac:26:   printf %s "(cached) " >&6
      configure.ac:26: else $as_nop
      configure.ac:26:   if test -n "$CC"; then
      configure.ac:26:   ac_cv_prog_CC="$CC" # Let the user override the test.
      configure.ac:26: else
      configure.ac:26: as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
      configure.ac:26: for as_dir in $PATH
      configure.ac:26: do
      configure.ac:26:   IFS=$as_save_IFS
      configure.ac:26:   case $as_dir in #(((
      configure.ac:26:     '' is m4_require'd but not m4_defun'd
      lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
      lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
      ./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
      ./lib/autoconf/programs.m4:41: _AC_CHECK_PROG is expanded from...
      ./lib/autoconf/programs.m4:101: AC_CHECK_PROG is expanded from...
      ./lib/autoconf/programs.m4:221: AC_CHECK_TOOL is expanded from...
      ./lib/autoconf/c.m4:452: AC_PROG_CC is expanded from...
      configure.ac:26: the top level
      Makefile.am:3: error: Libtool library used but 'LIBTOOL' is undefined
      Makefile.am:3:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
      Makefile.am:3:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
      Makefile.am:3:   If 'LT_INIT' is in 'configure.ac', make sure
      Makefile.am:3:   its definition is in aclocal's search path.
      autoreconf: error: automake failed with exit status: 1
      Using -march=native. Binary will not be portable.
      configure: WARNING: unrecognized options: --enable-shared, --with-pic
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a race-free mkdir -p... ./../../install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking whether to enable maintainer-specific portions of Makefiles... yes
      checking whether make supports the include directive... yes (GNU style)
      checking for arm64-apple-darwin20.0.0-gcc... arm64-apple-darwin20.0.0-clang
      checking whether the C compiler works... no
      configure: error: in `/Users/shamik/opt/NaMaster/_deps/libsharp2':
      configure: error: C compiler cannot create executables
      See `config.log' for more details
      make: *** No targets specified and no makefile found.  Stop.
      make: *** No rule to make target `install'.  Stop.
      cp: directory ../include/libsharp2 does not exist
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a thread-safe mkdir -p... ./install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking for style of include used by make... GNU
      checking for arm64-apple-darwin20.0.0-gcc... arm64-apple-darwin20.0.0-clang
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether arm64-apple-darwin20.0.0-clang accepts -g... yes
      checking for arm64-apple-darwin20.0.0-clang option to accept ISO C89... none needed
      checking whether arm64-apple-darwin20.0.0-clang understands -c and -o together... yes
      checking dependency style of arm64-apple-darwin20.0.0-clang... gcc3
      checking for arm64-apple-darwin20.0.0-ar... arm64-apple-darwin20.0.0-ar
      checking the archiver (arm64-apple-darwin20.0.0-ar) interface... ar
      checking build system type... Invalid configuration `arm64-apple-darwin20.0.0': machine `arm64-apple' not recognized
      configure: error: /bin/sh ./config.sub arm64-apple-darwin20.0.0 failed
      ERROR: failed to configure libnmt. Check all dependencies are installed
             Dependencies:
             - GSL
             - FFTW
             - CFITSIO
             - HEALPix
      error: Failed to compile C library.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pymaster

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.`

*Update 2*: No, linking the manually compiled libsharp2 libraries did not solve the problem. Seems like some of the previous issue linked to libsharp2 install and missing libtool on my system has been fixed. 
Error message:
```
Building wheels for collected packages: pymaster
  Building wheel for pymaster (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [35 lines of output]
      running bdist_wheel
      running build
      running build_py
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a thread-safe mkdir -p... ./install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking for style of include used by make... GNU
      checking for arm64-apple-darwin20.0.0-gcc... /Users/shamik/miniforge3/envs/arm/bin/clang
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether /Users/shamik/miniforge3/envs/arm/bin/clang accepts -g... yes
      checking for /Users/shamik/miniforge3/envs/arm/bin/clang option to accept ISO C89... none needed
      checking whether /Users/shamik/miniforge3/envs/arm/bin/clang understands -c and -o together... yes
      checking dependency style of /Users/shamik/miniforge3/envs/arm/bin/clang... gcc3
      checking for arm64-apple-darwin20.0.0-ar... arm64-apple-darwin20.0.0-ar
      checking the archiver (arm64-apple-darwin20.0.0-ar) interface... ar
      checking build system type... Invalid configuration `arm64-apple-darwin20.0.0': machine `arm64-apple' not recognized
      configure: error: /bin/sh ./config.sub arm64-apple-darwin20.0.0 failed
      ERROR: failed to configure libnmt. Check all dependencies are installed
             Dependencies:
             - GSL
             - FFTW
             - CFITSIO
             - HEALPix
      error: Failed to compile C library.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pymaster
  Running setup.py clean for pymaster
Failed to build pymaster
Installing collected packages: pymaster
  Running setup.py install for pymaster ... error
  error: subprocess-exited-with-error

  × Running setup.py install for pymaster did not run successfully.
  │ exit code: 1
  ╰─> [37 lines of output]
      running install
      /Users/shamik/miniforge3/envs/arm/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a thread-safe mkdir -p... ./install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking for style of include used by make... GNU
      checking for arm64-apple-darwin20.0.0-gcc... /Users/shamik/miniforge3/envs/arm/bin/clang
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether /Users/shamik/miniforge3/envs/arm/bin/clang accepts -g... yes
      checking for /Users/shamik/miniforge3/envs/arm/bin/clang option to accept ISO C89... none needed
      checking whether /Users/shamik/miniforge3/envs/arm/bin/clang understands -c and -o together... yes
      checking dependency style of /Users/shamik/miniforge3/envs/arm/bin/clang... gcc3
      checking for arm64-apple-darwin20.0.0-ar... arm64-apple-darwin20.0.0-ar
      checking the archiver (arm64-apple-darwin20.0.0-ar) interface... ar
      checking build system type... Invalid configuration `arm64-apple-darwin20.0.0': machine `arm64-apple' not recognized
      configure: error: /bin/sh ./config.sub arm64-apple-darwin20.0.0 failed
      ERROR: failed to configure libnmt. Check all dependencies are installed
             Dependencies:
             - GSL
             - FFTW
             - CFITSIO
             - HEALPix
      error: Failed to compile C library.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pymaster

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Update 3**:  @damonge @fjaviersanchez I think I have fixed this issue:
> It seems the current issue is
> 
> ```
> Invalid configuration `arm64-apple-darwin20.0.0': machine `arm64-apple' not recognized
> ```
> 
> I'm afraid I'm not proficient enough with mac to know what to do here. Perhaps @fjaviersanchez may have encountered this?

The configure file and config.sub files needed an addition of `arm64-apple-darwin*` option. In configure options for `darwin*` has been used `arm64-apple-darwin*`, while config.sub recognizes `arm64-apple` as `$basic_machine` and `-darwin*` as `$os`. 

Doing `pip install .` did not recognise fftw installed via `brew` and in `$PATH`. But fftw installed via conda is found. 

So I get a successful configure, but then I encounter a linking issue :
```
libtool: link: arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/shamik/miniforge3/envs/arm/include -fopenmp -O3 -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath -Wl,/Users/shamik/miniforge3/envs/arm/lib -o namaster src/namaster-nmt_main.o  -L./ -lnmt -lfftw3_omp -lm -L/Users/shamik/miniforge3/envs/arm/lib -L/Users/shamik/opt/NaMaster/_deps/lib -lsharp2 -lfftw3 -lchealpix -lcfitsio -lgsl -lgslcblas -fopenmp
      ld: library not found for -lnmt
      clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```
During configure the dependencies where all correctly foung:
```
      checking for library containing cblas_dgemm... -lgslcblas
      checking for library containing gsl_spline2d_eval_e... -lgsl
      checking for library containing ffopen... -lcfitsio
      checking for library containing ring2nest... -lchealpix
      checking for library containing fftw_execute... -lfftw3
      checking for library containing sharp_execute... -lsharp2
```
I am not sure what is causing the linking issue now. Any suggestion would be very helpful.

Full output
```
 Preparing metadata (setup.py) ... done
Building wheels for collected packages: pymaster
  Building wheel for pymaster (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [190 lines of output]
      running bdist_wheel
      running build
      running build_py
      checking for a BSD-compatible install... /usr/bin/install -c
      checking whether build environment is sane... yes
      checking for a race-free mkdir -p... ./install-sh -c -d
      checking for gawk... no
      checking for mawk... no
      checking for nawk... no
      checking for awk... awk
      checking whether make sets $(MAKE)... yes
      checking whether make supports nested variables... yes
      checking whether make supports the include directive... yes (GNU style)
      checking for arm64-apple-darwin20.0.0-gcc... arm64-apple-darwin20.0.0-clang
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether the compiler supports GNU C... yes
      checking whether arm64-apple-darwin20.0.0-clang accepts -g... yes
      checking for arm64-apple-darwin20.0.0-clang option to enable C11 features... none needed
      checking whether arm64-apple-darwin20.0.0-clang understands -c and -o together... yes
      checking dependency style of arm64-apple-darwin20.0.0-clang... gcc3
      checking for arm64-apple-darwin20.0.0-ar... arm64-apple-darwin20.0.0-ar
      checking the archiver (arm64-apple-darwin20.0.0-ar) interface... ar
      checking build system type... aarch64-apple-darwin20.0.0
      checking host system type... aarch64-apple-darwin20.0.0
      checking how to print strings... printf
      checking for a sed that does not truncate output... /usr/bin/sed
      checking for grep that handles long lines and -e... /usr/bin/grep
      checking for egrep... /usr/bin/grep -E
      checking for fgrep... /usr/bin/grep -F
      checking for ld used by arm64-apple-darwin20.0.0-clang... arm64-apple-darwin20.0.0-ld
      checking if the linker (arm64-apple-darwin20.0.0-ld) is GNU ld... no
      checking for BSD- or MS-compatible name lister (nm)... arm64-apple-darwin20.0.0-nm
      checking the name lister (arm64-apple-darwin20.0.0-nm) interface... BSD nm
      checking whether ln -s works... yes
      checking the maximum length of command line arguments... 786432
      checking how to convert aarch64-apple-darwin20.0.0 file names to aarch64-apple-darwin20.0.0 format... func_convert_file_noop
      checking how to convert aarch64-apple-darwin20.0.0 file names to toolchain format... func_convert_file_noop
      checking for arm64-apple-darwin20.0.0-ld option to reload object files... -r
      checking for arm64-apple-darwin20.0.0-file... no
      checking for file... file
      checking for arm64-apple-darwin20.0.0-objdump... no
      checking for objdump... objdump
      checking how to recognize dependent libraries... pass_all
      checking for arm64-apple-darwin20.0.0-dlltool... no
      checking for dlltool... no
      checking how to associate runtime and link libraries... printf %s\n
      checking for arm64-apple-darwin20.0.0-ar... (cached) arm64-apple-darwin20.0.0-ar
      checking for archiver @FILE support... no
      checking for arm64-apple-darwin20.0.0-strip... arm64-apple-darwin20.0.0-strip
      checking for arm64-apple-darwin20.0.0-ranlib... arm64-apple-darwin20.0.0-ranlib
      checking command to parse arm64-apple-darwin20.0.0-nm output from arm64-apple-darwin20.0.0-clang object... ok
      checking for sysroot... no
      checking for a working dd... /bin/dd
      checking how to truncate binary pipes... /bin/dd bs=4096 count=1
      checking for arm64-apple-darwin20.0.0-mt... no
      checking for mt... no
      checking if : is a manifest tool... no
      checking for arm64-apple-darwin20.0.0-dsymutil... no
      checking for dsymutil... dsymutil
      checking for arm64-apple-darwin20.0.0-nmedit... arm64-apple-darwin20.0.0-nmedit
      checking for arm64-apple-darwin20.0.0-lipo... arm64-apple-darwin20.0.0-lipo
      checking for arm64-apple-darwin20.0.0-otool... arm64-apple-darwin20.0.0-otool
      checking for arm64-apple-darwin20.0.0-otool64... no
      checking for otool64... no
      checking for -single_module linker flag... yes
      checking for -exported_symbols_list linker flag... yes
      checking for -force_load linker flag... yes
      checking for stdio.h... yes
      checking for stdlib.h... yes
      checking for string.h... yes
      checking for inttypes.h... yes
      checking for stdint.h... yes
      checking for strings.h... yes
      checking for sys/stat.h... yes
      checking for sys/types.h... yes
      checking for unistd.h... yes
      checking for dlfcn.h... yes
      checking for objdir... .libs
      checking if arm64-apple-darwin20.0.0-clang supports -fno-rtti -fno-exceptions... yes
      checking for arm64-apple-darwin20.0.0-clang option to produce PIC... -fno-common -DPIC
      checking if arm64-apple-darwin20.0.0-clang PIC flag -fno-common -DPIC works... yes
      checking if arm64-apple-darwin20.0.0-clang static flag -static works... no
      checking if arm64-apple-darwin20.0.0-clang supports -c -o file.o... yes
      checking if arm64-apple-darwin20.0.0-clang supports -c -o file.o... (cached) yes
      checking whether the arm64-apple-darwin20.0.0-clang linker (arm64-apple-darwin20.0.0-ld) supports shared libraries... yes
      checking dynamic linker characteristics... darwin20.0.0 dyld
      checking how to hardcode library paths into programs... immediate
      checking for dlopen in -ldl... yes
      checking whether a program can dlopen itself... yes
      checking whether a statically linked program can dlopen itself... yes
      checking whether stripping libraries is possible... yes
      checking if libtool supports shared libraries... yes
      checking whether to build shared libraries... yes
      checking whether to build static libraries... yes
      checking whether to enable maintainer-specific portions of Makefiles... no
      checking for arm64-apple-darwin20.0.0-gcc... (cached) arm64-apple-darwin20.0.0-clang
      checking whether the compiler supports GNU C... (cached) yes
      checking whether arm64-apple-darwin20.0.0-clang accepts -g... (cached) yes
      checking for arm64-apple-darwin20.0.0-clang option to enable C11 features... (cached) none needed
      checking whether arm64-apple-darwin20.0.0-clang understands -c and -o together... (cached) yes
      checking dependency style of arm64-apple-darwin20.0.0-clang... (cached) gcc3
      checking whether to enable maintainer-specific portions of Makefiles... no
      checking for arm64-apple-darwin20.0.0-clang option to support OpenMP... none needed
      checking for library containing pow... none required
      checking for library containing cblas_dgemm... -lgslcblas
      checking for library containing gsl_spline2d_eval_e... -lgsl
      checking for library containing ffopen... -lcfitsio
      checking for library containing ring2nest... -lchealpix
      checking for library containing fftw_execute... -lfftw3
      checking for library containing sharp_execute... -lsharp2
      checking for stdlib.h... (cached) yes
      checking for string.h... (cached) yes
      checking for math.h... yes
      checking for stdarg.h... yes
      checking for unistd.h... (cached) yes
      checking for time.h... yes
      checking for omp.h... yes
      checking for complex.h... yes
      checking for size_t... yes
      checking for GNU libc compatible malloc... (cached) yes
      checking for pow... yes
      checking for sqrt... yes
      checking that generated files are newer than configure... done
      configure: creating ./config.status
      config.status: creating Makefile
      config.status: creating config.h
      config.status: config.h is unchanged
      config.status: executing depfiles commands
      config.status: executing libtool commands

      -------------------------------------------------------
       namaster Version 0.0

       Configuration settings:

       - Prefix: '/Users/shamik/opt/NaMaster/_deps'
       - Compiler: 'arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/shamik/miniforge3/envs/arm/include -fopenmp -O3 -D_FORTIFY_SOURCE=2 -isystem /Users/shamik/miniforge3/envs/arm/include -I/Users/shamik/opt/NaMaster/_deps/include -fopenmp -O3'

       Now type 'make [<target>]', where the optional
       target may be:

         all     : to build everything (currently libnmt)
                   this is the default target
         check   : runs the library's unit tests
         install : to put everything in the installation
                   directories ('include', 'lib' and
               'bin' from prefix). Root privileges may
               be necessary for this.
         clean   : to remove garbage
      ------------------------------------------------------

      Successful configure.
       rm -f namaster
      test -z "libnmt.la" || rm -f libnmt.la
      rm -f ./so_locations
      rm -rf .libs _libs
      rm -rf src/.libs src/_libs
      rm -rf test/.libs test/_libs
       rm -f test/check_nmt
      rm -f *.o
      rm -f src/*.o
      rm -f src/*.lo
      rm -f test/*.o
      rm -f *.lo
      /Library/Developer/CommandLineTools/usr/bin/make  all-am
      arm64-apple-darwin20.0.0-clang -DHAVE_CONFIG_H -I.   -I./src/ -D_FORTIFY_SOURCE=2 -isystem /Users/shamik/miniforge3/envs/arm/include -I/Users/shamik/opt/NaMaster/_deps/include -fopenmp -O3  -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/shamik/miniforge3/envs/arm/include -fopenmp -O3 -MT src/namaster-nmt_main.o -MD -MP -MF src/.deps/namaster-nmt_main.Tpo -c -o src/namaster-nmt_main.o `test -f 'src/nmt_main.c' || echo './'`src/nmt_main.c
      mv -f src/.deps/namaster-nmt_main.Tpo src/.deps/namaster-nmt_main.Po
      /bin/sh ./libtool  --tag=CC   --mode=link arm64-apple-darwin20.0.0-clang  -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/shamik/miniforge3/envs/arm/include -fopenmp -O3  -L./ -lnmt -lfftw3_omp -lfftw3 -lgsl -lgslcblas -lcfitsio -lsharp2 -lm -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/shamik/miniforge3/envs/arm/lib -L/Users/shamik/miniforge3/envs/arm/lib -L/Users/shamik/opt/NaMaster/_deps/lib -o namaster src/namaster-nmt_main.o  -lsharp2 -lfftw3 -lchealpix -lcfitsio -lgsl -lgslcblas
      libtool: link: arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/shamik/miniforge3/envs/arm/include -fopenmp -O3 -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath -Wl,/Users/shamik/miniforge3/envs/arm/lib -o namaster src/namaster-nmt_main.o  -L./ -lnmt -lfftw3_omp -lm -L/Users/shamik/miniforge3/envs/arm/lib -L/Users/shamik/opt/NaMaster/_deps/lib -lsharp2 -lfftw3 -lchealpix -lcfitsio -lgsl -lgslcblas -fopenmp
      ld: library not found for -lnmt
      clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
      make[1]: *** [namaster] Error 1
      make: *** [all] Error 2
      ERROR: couldn't compile namaster. Make sure all dependencies are accessible.
             You may need to add the correct paths to CPPFLAGS, LDFLAGS and LD_LIBRARY_PATHS.
             E.g.:
               >$ export CPPFLAGS+=" -I/path/to/deps/include"
               >$ export LDFLAGS+=" -L/path/to/deps/lib"
               >$ export LD_LIBRARY_PATHS=$LD_LIBRARY_PATH:/path/to/deps/lib

             Dependencies:
             - GSL
             - FFTW
             - CFITSIO
             - HEALPix
      error: Failed to compile C library.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pymaster
  Running setup.py clean for pymaster
Failed to build pymaster
...
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -9,6 +9,7 @@ include_HEADERS = src/namaster.h
 
 bin_PROGRAMS = namaster
 namaster_SOURCES = src/nmt_main.c
+namaster_LDADD = libnmt.la
 namaster_CPPFLAGS = $(OPENMP_CFLAGS) -I./src/
 namaster_LDFLAGS = $(OPENMP_CFLAGS) -L./ -lnmt @fftw_thread_type@ -lfftw3 -lgsl -lgslcblas -lcfitsio @libsharp_libs@ -lm
 

